staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -402,16 +466,50 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 	if !strings.HasSuffix(key, "/") {
 		key += "/"
 	}
-	getResp, err := s.client.KV.Get(ctx, key, clientv3.WithPrefix())
+	keyPrefix := key
+
+	// set the appropriate clientv3 options to filter the returned data set
+	options := make([]clientv3.OpOption, 0, 4)
+	if pred.Limit > 0 {
+		options = append(options, clientv3.WithLimit(pred.Limit))
+	}
+	var returnedRV int64
+	switch {
+	case len(pred.From) > 0:
+		fromKey, fromRV, err := decodeFrom(pred.From, keyPrefix)
+		if err != nil {
+			return err
+		}
+
+		options = append(options, clientv3.WithRange(key+"\xFF"))
```

Comment:
Thanks for clarifying. I did find `From` a bit difficult to grok at first, so agree that name like `Continue` or `NextPageToken` might be helpful.